docs/specs/shapes.md

```diff
@@ -7,6 +7,52 @@ The graphical elements are divided in 4 categories:
 * [Styles](#shape-style), that define the visual appearance of shapes
 * [Modifiers](#modifier) alter the curves of the shapes
 
+## Shape Rendering Model
+
+### Grouping and Ordering Rules
+
+* **shapes** are rendered in reverse order (bottom->top)
+* **groups** offer a scoping mechanism for transforms, styles, modifiers, and shapes
+* **transforms** adjust the coordinate system for all elements within their group, and transitively
+  for all other group-nested elements
+* **styles** and **modifiers** apply to all preceding shapes within the current scope,
+  including group-nested shapes
+* when **multiple styles** apply to the same shape, the shape is rendered repeatedly for each style,
+  in reverse order
+* when **multiple modifiers** apply to the same shape, they are composed in reverse order
+  (e.g. $Trim(Trim(shape))$)
+* when **multiple transforms** apply to the same shape (due to scope nesting), they compos in group
```

Comment:
Composing means the transforms are additive (instead of being replacements)?

Typo: compos